### PR TITLE
Viite 113 check entire road network

### PR DESCRIPTION
**Pull request by plourenco90**

### CODE REVIEW

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala [hunk 1]
```diff
@@ -1178,4 +1192,27 @@ object RoadAddressDAO {
...
+    val query = s"""select ra.id, ra.road_number, ra.road_part_number, ra.road_type, ra.track_code,
+        ra.discontinuity, ra.start_addr_m, ra.end_addr_m, ra.lrm_position_id, pos.link_id, pos.start_measure, pos.end_measure,
+        pos.side_code, pos.adjusted_timestamp,
+        ra.start_date, ra.end_date, ra.created_by, ra.valid_from, ra.CALIBRATION_POINTS, ra.floating, t.X, t.Y, t2.X, t2.Y, link_source, ra.ely, ra.terminated
+        from road_address ra cross join
+        TABLE(SDO_UTIL.GETVERTICES(ra.geometry)) t cross join
+        TABLE(SDO_UTIL.GETVERTICES(ra.geometry)) t2
+        join lrm_position pos on ra.lrm_position_id = pos.id
+        where t.id < t2.id and
+          (valid_from is null or valid_from <= sysdate) and
```
Comment (oappicgi) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala]: valid_from cannot be nullable in roadaddress table

**Comment by oappicgi** [comment]
backend check would be nice, code seems fine

File: digiroad2-api-common/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Context.scala [hunk 2]
```diff
@@ -166,6 +166,13 @@ class RoadAddressFloater(roadAddressService: RoadAddressService) extends Actor {
   }
 }
 
+class RoadNetworkChecker(roadNetworkService: RoadNetworkService) extends Actor {
+  def receive = {
+    case w: Int => roadNetworkService.checkRoadAddressNetwork(w.asInstanceOf[Option[Long]])
```
Comment (pedrosag) [inline digiroad2-api-common/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Context.scala]: Can this case w: Int be switched to case w: Long, therefore avoiding the type cast in the end of the line?

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  def checkRoadAddressNetwork(roadNumber: Option[Long]): Unit = {
+
+    def checkCalibrationPoints(road1: RoadAddress, road2: RoadAddress): Boolean = {
+      road1.track != road2.track && road1.calibrationPoints._1.isEmpty && road1.calibrationPoints._2.isEmpty &&
+        road2.calibrationPoints._1.isEmpty && road2.calibrationPoints._2.isEmpty
+    }
+
+    def checkOverlapping(road1: RoadAddress, road2: RoadAddress) = {
+      road1.endAddrMValue != road2.startAddrMValue && road1.track.value == road2.track.value match {
```
Comment (pedrosag) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala]: Another question: Does could this benefit from the overlaps method on the GeometryUtils or is it a completely different purpose?

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        case true => RoadNetworkDAO.addRoadNetworkError(road1.id, OverlappingRoadAddresses.value)
+        case _ => None
+      }
+    }
+
+    def checkTopology(road1: RoadAddress, road2: RoadAddress) = {
+      !GeometryUtils.areAdjacent(road1.geometry, road2.geometry) &&
+        (road1.discontinuity != Discontinuity.MinorDiscontinuity || road1.discontinuity != Discontinuity.Discontinuous ||
```
Comment (Campanha) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadNetworkService.scala]: we have some repeated validation

**Pull request merged**